sympy/plotting/plot.py

```diff
@@ -2192,7 +2228,17 @@ def check_arguments(args, expr_len, nb_of_free_symbols):
 
     elif isinstance(args[0], Tuple) and len(args[0]) == expr_len + nb_of_free_symbols:
         #Multiple plots with different ranges.
+        lst = []
         for arg in args:
+            #Warn: Dimension
+            # get all variables
+            sbls = sum(arg[:expr_len]).free_symbols
+            # get symbols from all ranges
+            specify_symbols = set([s[0] for s in arg[expr_len:]])
+            if len(sbls) != nb_of_free_symbols or specify_symbols-sbls:
```

Comment:
One space before and after -.

Actually what does the line above mean? Should it be:
```python
            if len(sbls) != nb_of_free_symbols and len(sbls) != specify_symbols - sbls:
```
which could be instead
```python
            if len(sbls) not in (nb_of_free_symbols, specify_symbols - sbls):
```

Or should it be
```python
            if len(sbls) != nb_of_free_symbols or specify_symbols - sbls != 0:
```